virtcontainers/api.go

```diff
@@ -122,6 +122,11 @@ func createSandboxFromConfig(ctx context.Context, sandboxConfig SandboxConfig, f
 		return nil, err
 	}
 
+	// Setup host cgroups
+	if err := s.setupCgroups(); err != nil {
```

Comment:
This is already called in `createContainers()`. Are you trying to make sure there is host cpu cg even for an empty sandbox?

Reply:
No, this `s.setupCgroups()` in `createSandboxFromConfig` is for the initial sandbox container, in this sandbox it has exactly one container. Another call of `s.setupCgroups()` is in `s.CreateContainer()`. The setup process only happens once for each container(including sandbox container), there's no duplicate call.

Reply:
@WeiZhang555 There is a code path `createSandboxFromConfig` -> `s.createContainers` -> `createContainer` -> `setupCgroups` for the initial container in the sandbox as well.

Reply:
`createContainer` -> `setupCgroups` doesn't exist.

Maybe you mixed `createContainer()` from `virtcontainers/container.go` with the one from `cli/create.go` ?

Reply:
oops, you are right! I mixed the two createcontainer() functions in sandbox.go and container.go, though they have a initial capital difference. Sry for the noise...